### DISCUSSION TITLE
Better UI/UX

### DISCUSSION THREAD

**Discussion by AlixShahid** [mistweaverco/kulala.nvim#146] [Ideas]
The following is an example:

```http
# book.http
@TAGS=Book

# user.http
@TAGS=Authentication, Book
```

The files `book.http` and `user.http` have tags at the top, and based on these tags, we can organize the requests.

Kulala will feature a three-split window layout:
1. Organized requests or all available HTTP files in the project.
2. File window.
3. Response window.

Each window will have specific key bindings. For example, pressing `H` will show the header, and `B` will display the body.

Additionally, we will cache the last request, allowing users to retrieve the last response of a specific endpoint.

![bitmap](https://github.com/user-attachments/assets/083422ed-e549-4664-80ab-b0038d0593f7)

**Comment by gorillamoe** [comment]
Looks promising to me, but as with all the features, it should be opt-in, not opt-out.

Re-running the last request ist already implemented via `.replay()`.

One minor change I would like is tagging it via comments at the top:

```http
# @tags foo, bar, Naz

GET https://httpbin.org/get HTTP/1.1
```

**Comment by LintaoAmons** [comment]
```
require('kulala').toggle_view()
```

I notice we have this command to switch between response header and body, but maybe it's better to just have both of them show in the result panels

something like 

<img width="975" alt="image" src="https://github.com/user-attachments/assets/9340c787-3727-4ec6-abab-8c5273b07e63">

**Reply by gorillamoe** [reply]
> ```
> require('kulala').toggle_view()
> ```
> 
> I notice we have this command to switch between response header and body, but maybe it's better to just have both of them show in the result panels


Already configurable 👌🏾

https://kulala.mwco.app/docs/getting-started/configuration-options#default_view
